test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1298,6 +1300,45 @@
             (Thread/sleep 100))
           (is (= {:received-commands 1 :executed-commands 1} (stats))))))))
 
+(deftest date-round-trip
+  (svc-utils/with-puppetdb-instance
+    (let [pdb (get-service svc-utils/*server* :PuppetDBServer)
+          dispatcher (get-service svc-utils/*server* :PuppetDBCommandDispatcher)
+          shared-globals (partial cli-svc/shared-globals pdb)
+          enqueue-command (partial enqueue-command dispatcher)
+          globals (shared-globals)
+          {{:keys [connection endpoint]} :command-mq} globals
+          deactivate-ms 14250331086887
+          ;; The problem only occurred if you passed a Date to
+          ;; enqueue, a DateTime wasn't a problem.
+          input-stamp (java.util.Date. deactivate-ms)
+          expected-stamp (DateTime. deactivate-ms DateTimeZone/UTC)]
+      (enqueue-command connection endpoint
+                       (command-names :deactivate-node) 3
+                       {:certname "foo.local" :producer_timestamp input-stamp})
+      (svc-utils/wait-for-server-processing svc-utils/*server*)
+      ;; While we're here, check the value in the database too...
+      (is (= expected-stamp
+             (jdbc/with-transacted-connection
+               (:scf-read-db globals)
+               :repeatable-read
+               (from-sql-date (scf-store/node-deactivated-time "foo.local")))))
```

Comment:
If I understand what you're asking correctly, I just added this because it was easy, and tested exactly what was in the DB without the influence of any extra translations on the way out.  It *might* provide a bit of extra safety, but we don't have to have it.